imagepipeline/src/main/java/com/facebook/imagepipeline/core/ProducerSequenceFactory.java

```diff
@@ -192,8 +192,12 @@ private static void validateEncodedImageRequest(ImageRequest imageRequest) {
     } else if (UriUtil.isLocalResourceUri(uri)) {
       return getLocalResourceFetchSequence();
     } else {
+      String uriString = uri.toString();
+      if (!TextUtils.isEmpty(uriString)) {
+        uriString.substring(0, Math.min(30, uriString.length));
+      }
       throw new RuntimeException(
-          "Unsupported image type! Uri is: " + uri.toString().substring(0, 30));
+          "Unsupported image type! Uri is: " + uriString);
```

Comment:
This is really "uri scheme", not "image type" so we should change the message too.